src/detect-cipservice.c

```diff
@@ -0,0 +1,504 @@
+
+    s->alproto = ALPROTO_ENIP;
+
+    SigMatchAppendSMToList(s, sm, DETECT_SM_LIST_CIP_MATCH);
+
+    SCReturnInt(0);
+
+    error: if (cipserviced != NULL)
+        DetectCipServiceFree(cipserviced);
+    if (sm != NULL)
+        SCFree(sm);
+    printf("DetectCipServiceSetup - Error\n");
+
+    SCReturnInt(-1);
+}
+
+/**
+ * \brief this function will free memory associated with DetectCipServiceData
+ *
+ * \param ptr pointer to DetectCipServiceData
+ */
+void DetectCipServiceFree(void *ptr)
+{
+    DetectCipServiceData *cipserviced = (DetectCipServiceData *) ptr;
+    SCFree(cipserviced);
+}
+
+#ifdef UNITTESTS
+
+/**
+ * \test Test CIP Command parameter parsing
+ */
+static int DetectCipServiceParseTest01 (void)
+{
+
+    uint8_t res = 1;
+
+    /*DetectCipServiceData *cipserviced = NULL;
+     cipserviced = DetectCipServiceParse("1");
+     if (cipserviced != NULL)
+     {
+     if (cipserviced->cipservice == 1)
+     {
+     res = 1;
+     }
+
+     DetectCipServiceFree(cipserviced);
+     }
+     */
+    return res;
+}
+
+/**
+ * \test Test CIP Service signature
+ */
+static int DetectCipServiceSignatureTest01 (void)
+{
+    uint8_t res = 0;
+
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    if (de_ctx == NULL)
+    goto end;
+
+    Signature *sig = DetectEngineAppendSig(de_ctx, "alert tcp any any -> any any (cip_service:1; sid:1; rev:1;)");
+    if (sig == NULL)
+    {
+        printf("parsing signature failed: ");
+        goto end;
+    }
+
+    /* if we get here, all conditions pass */
+    res = 1;
+    end:
+    if (de_ctx != NULL)
+    DetectEngineCtxFree(de_ctx);
+    return res;
+}
+
+#endif /* UNITTESTS */
+
+/**
+ * \brief this function registers unit tests for DetectCipService
+ */
+void DetectCipServiceRegisterTests(void)
+{
+#ifdef UNITTESTS
+    UtRegisterTest("DetectCipServiceParseTest01",
+            DetectCipServiceParseTest01);
+    UtRegisterTest("DetectCipServiceSignatureTest01",
+            DetectCipServiceSignatureTest01);
+#endif /* UNITTESTS */
+}
+
+/*
+ *
+ ************************************************************ ENIP COMMAND CODE ********************************************************************
+ *
+ */
+
+/**
+ * \brief ENIP Commond Detect Prototypes
+ */
+static int DetectEnipCommandSetup(DetectEngineCtx *, Signature *, char *);
+static void DetectEnipCommandFree(void *);
+static void DetectEnipCommandRegisterTests(void);
+
+/**
+ * \brief Registration function for enip_command: keyword
+ */
+void DetectEnipCommandRegister(void)
+{
+    sigmatch_table[DETECT_ENIPCOMMAND].name = "enip_command"; //rule keyword
+    sigmatch_table[DETECT_ENIPCOMMAND].desc
+            = "Rules for detecting EtherNet/IP command";
+    sigmatch_table[DETECT_ENIPCOMMAND].url = "www.solananetworks.com";
+    sigmatch_table[DETECT_ENIPCOMMAND].Match = NULL;
+    sigmatch_table[DETECT_ENIPCOMMAND].AppLayerMatch = NULL;
+    sigmatch_table[DETECT_ENIPCOMMAND].alproto = ALPROTO_ENIP;
```

Comment:
These generate compiler warnings, as can be seen in the failed travis ci test. Please don't ignore ANY compiler warning.